From e077a85453c6f93625526d74902e3ee8c3fdec17 Mon Sep 17 00:00:00 2001
From: "Christopher A. Williamson" <contact@chrisaw.io>
Date: Fri, 27 Mar 2026 11:02:39 +0000
Subject: [PATCH 02/24] guard goodix sysfs write behind existence check

avoid errors on devices without the goodix_ts driver by checking for
/sys/devices/platform/goodix_ts.0/support_pen before writing to it.
---
 rw-system.sh | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/rw-system.sh b/rw-system.sh
index 3ac9aa5..1fb1e40 100644
--- a/rw-system.sh
+++ b/rw-system.sh
@@ -1247,4 +1247,6 @@ if getprop ro.vendor.build.fingerprint | grep -iq -e Rebecco/K70_ROW; then
 fi
 
 # Enable pen mode on Lenovo/goodix
-echo 1 > /sys/devices/platform/goodix_ts.0/support_pen
+if [ -e /sys/devices/platform/goodix_ts.0/support_pen ]; then
+    echo 1 > /sys/devices/platform/goodix_ts.0/support_pen
+fi
-- 
2.53.0

